aspnet/mvc/views/forms/sample/final/Controllers/DemoRegisterController.cs

```diff
@@ -0,0 +1,71 @@
+﻿using FormsTH.ViewModels;
+using Microsoft.AspNet.Mvc;
+
+namespace FormsTH.Controllers
+{
+    // This controller is used only to demonstrate working with forms.
+    public class DemoRegController : Controller
+    {
+        static string _message;
+        public IActionResult Register()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Register(RegisterViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                _message = "HttpPost Register success";
```

Comment:
Do not demonstrate such a poor practise. Simply cannot work when multiple users interact with this site. Pass in a `routeValues` object if absolutely necessary. But it's not clear why this is worthwhile in the first place. Suggest just removing.